spec/controllers/campaigns_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe CampaignsController, type: :controller do
+  let(:campaign) { create(:campaign) }
+
+  describe 'GET #index' do
+    it 'renders index template' do
+      get :index
+      is_expected.to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'renders show template' do
+      get :show, params: { id: campaign.id }
+      is_expected.to render_template(:show)
+    end
+  end
+
+  describe 'GET #new' do
+    it 'renders new template' do
+      get :new
+      is_expected.to render_template(:new)
+    end
+  end
+
+  describe 'POST #create' do
```

Comment:
Same here

Also you don't cover the case when the object was not created

Reply:
added